### PR TITLE
Policy wise grouping feature 

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: modules/jaggery-apps/app-publisher-web/apis/v1/entitlement_api_router.jag [hunk 1]
```diff
@@ -50,6 +52,12 @@ require('/modules/publisher.js').exec(function(ctx) {
         print(partialList);
     });
 
+    routeManager.register('GET', 'publisher', '/publisher/api/entitlement/get/policy/Group/by/appId/{applicationId}', function (context) {
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/apis/v1/entitlement_api_router.jag]: This URL mapping is bad. Please explain the reason behind using this style.

File: modules/jaggery-apps/app-publisher-web/themes/appm/helpers/view-asset.js [hunk 2]
```diff
@@ -29,15 +29,42 @@ var merge = function(data){
...
+		 } else if (data.data.fields[i].name == 'oauthapis_webappConsumerSecret')
 			 data.data.fields[i].value = sResult.split(",")[1];
 
+
+		 for (var c in data.data.fields) {
+
+			 if (data.data.fields[i].name == "uritemplate_policygroupid" + c) {
+				 if (data.data.fields[i].value != "" && !isNaN(data.data.fields[i].value)) {
+                     //replace group policy id by name
+                     for(var j=0; j < resultPolicy.length; j++){
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/helpers/view-asset.js]: Formatting issues.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js [hunk 3]
```diff
@@ -65,7 +76,7 @@ $(document).on("click", "#btn-policy-save", function () {
     var policyName = $('#entitlement-policy-editor #policy-name').val();
 
     if(policyContent == "" || policyName == ""){
-        showEntitlementError("fields cannot be blank");
+        showEntitlementNotification("fields cannot be blank", "alert-danger");
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js]: Capitalize the alert message.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js [hunk 4]
```diff
@@ -80,7 +91,7 @@ $(document).on("click", "#btn-policy-partial-validate", function () {
...
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js]: Capitalize the alert message.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js [hunk 5]
```diff
@@ -98,7 +109,7 @@ $(document).on("click", "#btn-policy-save-and-close", function () {
...
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js]: Capitalize the alert message.

File: modules/jaggery-apps/app-publisher-web/apis/v1/entitlement_api_router.jag (hunk 1 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/apis/v1/entitlement_api_router.jag]: There was an issue with the url router. (if there are more than one url with same no of parameters, it referred the first url)
So till we fix the router issue I had to use this pattern to make the code work.

File: modules/jaggery-apps/app-publisher-web/themes/appm/helpers/view-asset.js (hunk 2 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/helpers/view-asset.js]: commit# https://github.com/lahiru823/carbon-appmgt/commit/e57c1e0375ae2bc3ade783846234975960942e2c

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js (hunk 3 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js]: this js is no longer used in the code.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js (hunk 5 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js]: This js is no longer used in the code.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js (hunk 4 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/entitlement.js]: This js is no longer used in the code.

**Pull request merged**